File: web/vue/src/ai/types/modelConfig.ts
/**
 * 模型配置类型定义
 */

/** 模型类型 */
export type ModelTypeKey =
  | "llm"
  | "text-embedding"
  | "rerank"
  | "speech2text"
  | "moderation"
  | "tts"
  | "text2img";

/** 模型类型标签映射 */
export const MODEL_TYPE_LABELS: Record<ModelTypeKey, string> = {
  llm: "LLM",
  "text-embedding": "嵌入",
  rerank: "排序",
  speech2text: "语音转文本",
  moderation: "内容审核",
  tts: "文本转语音",
  text2img: "文本转图片",
};

/** 模型配置项（二级行数据） */
export interface ModelConfigItem {
  model_name: string;
  model_label?: string;
  model_type: ModelTypeKey;
  is_default: boolean;
}

/** 插件及其模型（一级+二级数据） */
export interface PluginWithModels {
  plugin_id: string;
  plugin_name: string;
  status: string;
  models: ModelConfigItem[];
}

/** 默认模型展示项 */
export interface DefaultModelItem {
  model_type: ModelTypeKey;
  plugin_id?: string;
  model_name?: string;
  is_valid: boolean;
}

/** 可选模型项（配置模型弹窗用） */
export interface AvailableModelItem {
  model_name: string;
  model_label?: string;
  model_type: ModelTypeKey;
  is_enabled: boolean;
}

/** 默认模型选择项（设置默认弹窗用） */
export interface ModelSelectItem {
  plugin_id: string;
  plugin_name: string;
  provider: string;
  model_name: string;
  model_label?: string;
  model_type: ModelTypeKey;
}

/** 模型配置页面聚合响应 */
export interface ModelConfigOverviewResponse {
  total_plugins: number;
  configured_plugins: number;
  total_models: number;
  default_models: DefaultModelItem[];
  plugins: PluginWithModels[];
}

File: server/python/src/ai/schemas/model_config.py
"""模型配置 Schema"""

from __future__ import annotations

from framework.schemas import BaseModel


class ModelConfigItemResponse(BaseModel):
    """模型配置项响应"""

    model_name: str
    model_label: str | None = None
    model_type: str
    is_default: bool = False


class PluginWithModelsResponse(BaseModel):
    """插件及其模型列表响应"""

    plugin_id: str
    plugin_name: str
    status: str
    models: list[ModelConfigItemResponse] = []


class DefaultModelItemResponse(BaseModel):
    """默认模型展示项响应"""

    model_type: str
    plugin_id: str | None = None
    model_name: str | None = None
    is_valid: bool = True


class ModelConfigOverviewResponse(BaseModel):
    """模型配置页面聚合响应"""

    total_plugins: int = 0
    configured_plugins: int = 0
    total_models: int = 0
    default_models: list[DefaultModelItemResponse] = []
    plugins: list[PluginWithModelsResponse] = []


class AvailableModelItemResponse(BaseModel):
    """插件可用模型项响应"""

    model_name: str
    model_label: str | None = None
    model_type: str
    is_enabled: bool = False


class PluginAvailableModelsResponse(BaseModel):
    """插件可用模型列表响应"""

    models: list[AvailableModelItemResponse] = []


class ModelSelectItemResponse(BaseModel):
    """默认模型选择项响应"""

    plugin_id: str
    plugin_name: str
    provider: str
    model_name: str
    model_label: str | None = None
    model_type: str


class ModelSelectListResponse(BaseModel):
    """默认模型选择列表响应"""

    models: list[ModelSelectItemResponse] = []


class DefaultModelItemRequest(BaseModel):
    """批量设置默认模型项请求"""

    model_type: str
    plugin_id: str
    model_name: str | None = None


class BatchSetDefaultModelRequest(BaseModel):
    """批量设置默认模型请求"""

    items: list[DefaultModelItemRequest]


class EnabledModelsRequest(BaseModel):
    """配置启用模型请求"""

    model_names: list[str]
